src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadSet.cs

```diff
@@ -81,6 +81,45 @@ public static WorkloadSet FromJson(string json, SdkFeatureBand defaultFeatureBan
 #endif
         }
 
+        public static WorkloadSet FromWorkloadSetFolder(string path, string workloadSetVersion, SdkFeatureBand defaultFeatureBand)
+        {
+            WorkloadSet? workloadSet = null;
+            foreach (var jsonFile in Directory.GetFiles(path, "*.workloadset.json"))
+            {
+                var newWorkloadSet = WorkloadSet.FromJson(File.ReadAllText(jsonFile), defaultFeatureBand);
+                if (workloadSet == null)
+                {
+                    workloadSet = newWorkloadSet;
+                }
+                else
+                {
+                    //  If there are multiple workloadset.json files, merge them
+                    foreach (var kvp in newWorkloadSet.ManifestVersions)
+                    {
+                        if (workloadSet.ManifestVersions.ContainsKey(kvp.Key))
+                        {
+                            throw new InvalidOperationException($"Workload set files in {path} defined the same manifest ({kvp.Key}) multiple times");
+                        }
+                        workloadSet.ManifestVersions.Add(kvp.Key, kvp.Value);
+                    }
+                }
+            }
+
+            if (workloadSet == null)
+            {
+                throw new InvalidOperationException("No workload set information found in: " + path);
+            }
+
+            if (File.Exists(Path.Combine(path, "baseline.workloadset.json")))
+            {
+                workloadSet.IsBaselineWorkloadSet = true;
+            }
+
+            workloadSet.Version = workloadSetVersion;
```

Comment:
It feels a little weird to assert that this random workload set we found is the version specified in the argument to this method...maybe move this to be next to where we actually figured out what workload set version we wanted? Note that I think if we request version x from NuGet, we might get something else that's close.